src/domain/model/authentication/AuthenticatorFactory.ts

```diff
@@ -0,0 +1,24 @@
+import { Authenticator } from './Authenticator';
+import { AuthenticationMethod } from '../user/AuthenticationMethod';
+import { InvalidDomainStateError } from '../InvalidDomainStateError';
+
+export class AuthenticatorFactory {
+  private authenticators: Map<AuthenticationMethod, Authenticator>;
+
+  constructor(authenticators: Authenticator[]) {
+    this.authenticators = new Map(authenticators.map((authenticator) => [authenticator.handles, authenticator]));
+    this.checkAllAuthenticationMethodsImplemented();
```

Comment:
Can't we use typescript/interfaces to ensure this?

Reply:
It is injected at runtime